towhee/hub/operator_manager.py

```diff
@@ -0,0 +1,172 @@
+# Copyright 2021 Zilliz. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import requests
+import random
+import subprocess
+import sys
+import yaml
+from pathlib import Path
+from importlib import import_module
+
+from towhee.hub.download_tools import obtain_lfs_extensions, latest_commit, get_file_list, download_files
+from towhee.hub.repo_manager import RepoManager
+from towhee.utils.log import engine_log
+from requests.exceptions import HTTPError
+
+
+class OperatorManager(RepoManager):
+    """
+    The Repo Manager to manage the operator repos.
+
+    Args:
+        author (`str`):
+            The author of the repo.
+        repo (`str`):
+            The name of the repo.
+    """
+    def __init__(self, author: str, repo: str):
+        super().__init__(author, repo)
+        self._repo_type = 'operator'
+
+    def create(self, password: str, private: bool = True) -> None:
+        """
+        Create a repo under the account connected to the given token.
+
+        Args:
+            password (`str`):
+                Current author's password.
+            private (`bool`):
+                If the repo is private.
+
+        Raises:
+            (`HTTPError`)
+                Raise error in request.
+        """
+        if self.exists():
+            engine_log.info('%s/%s repo already exists.', self._author, self._repo)
+            return None
+
+        token_name = random.randint(0, 10000)
+        token_id, token_hash = self.create_token(token_name, password)
+
+        data = {
+            'auto_init': True,
+            'default_tag': 'main',
+            'description': 'This is another operator repo',
+            'name': self._repo,
+            'private': private,
+            'template': False,
+            'trust_model': 'default',
+            'type': self._type_dict[self._repo_type]
+        }
+        url = 'https://hub.towhee.io/api/v1/user/repos'
+
+        try:
+            r = requests.post(url, data=data, headers={'Authorization': 'token ' + token_hash})
+            r.raise_for_status()
+        except HTTPError as e:
+            self.delete_token(token_id, password)
+            raise e
+
+        self.delete_token(token_id, password)
+
+    def init(self) -> None:
```

Comment:
Can you add a `tag` parameter? And pass this function, because there is another PR to complete it #425 .